naturalspeech2_pytorch/naturalspeech2_pytorch.py

```diff
@@ -1043,6 +1054,13 @@ def __init__(
         scale = 1.                             # this will be set to < 1. for better convergence when training on higher resolution images
     ):
         super().__init__()
+
+        self.phoneme_enc = PhonemeEncoder(tokenizer=tokenizer)
+        self.prompt_enc = SpeechPromptEncoder(dim_codebook=128) # 128 for encodec
```

Comment:
in the following 4 lines, all the hard coded number (128, 512, 80, 256), should be configurable from the `NaturalSpeech2` init